vite.config.ts

```diff
@@ -2,14 +2,28 @@ import { defineConfig } from 'vite'
 import react from '@vitejs/plugin-react'
 
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(({ command }) => ({
   plugins: [react()],
+  root: 'src',
+  base: command === 'serve' ? '' : '/dist/',
+  build: {
+    // output dir for production build
+    outDir: '../dist',
+    emptyOutDir: true,
+    // emit manifest so PHP can find the hashed files
+    manifest: true,
+    // our entry
+    rollupOptions: {
+      input: './src/main.tsx',
+    }
+  },
 
   server: {
     host: "0.0.0.0",
+    origin: 'http://vite-project.ddev.site:5273',
```

Comment:
This gets the react logo working. That file is loaded and processed through the Vite dev server.

However... the Vite logo is using the `public` directory. I don't have that working at the moment.